server/src/main/java/org/elasticsearch/indices/breaker/HierarchyCircuitBreakerService.java

```diff
@@ -739,7 +739,7 @@ private TriggerGCResult tryTriggerGC(MemoryUsage memoryUsed) {
                     if (initialCollectionCount != gcCountSupplier.getAsLong()) {
                         break;
                     }
-                    localBlackHole += new byte[allocationSize].hashCode();
+                    localBlackHole += System.identityHashCode(new byte[allocationSize]);
```

Comment:
This is basically the same as before, but clearly communicates the intention instead of looking as an oversight.

Reply:
I don't think this needs to be changed - we're not collecting the hash code to do anything, just making sure the allocation doesnt get optimised away. I think we just need a comment here saying this is to check for an allocation, and the object is being used so the JVM doesnt prematurely remove it